### PR TITLE
Add namespace to cluster resource.

### PR DESCRIPTION
**Pull request by cappyzawa**
Signed-off-by: cappyzawa <cappyzawa@yahoo.ne.jp>

# Changes

I expect this to reduce args passed to `kubectl apply`.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) and [TaskRun](../tekton/publish-run.yaml) to build and release this image

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md)
are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes)
must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS)
and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes)
must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS),
and they must first be added
[in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Enable to use `namespace` in ClusterResource params.
```

### CODE REVIEW

File: docs/resources.md [hunk 1]
```diff
@@ -356,6 +356,7 @@ The Cluster resource has the following parameters:
 -   `url` (required): Host url of the master node
 -   `username` (required): the user with access to the cluster
 -   `password`: to be used for clusters with basic auth
+-   `namespace`: The namespace to be given to the target namespace
```
Comment (sbwsg) [inline docs/resources.md]: Suggest changing this to `The namespace to target in the cluster`

File: cmd/kubeconfigwriter/main.go [hunk 2]
```diff
@@ -81,8 +81,9 @@ func createKubeconfigFile(resource *v1alpha1.ClusterResource, logger *zap.Sugare
 		Password: pass,
 	}
 	context := &clientcmdapi.Context{
-		Cluster:  resource.Name,
-		AuthInfo: resource.Username,
+		Cluster:   resource.Name,
+		AuthInfo:  resource.Username,
+		Namespace: resource.Namespace,
```
Comment (sbwsg) [inline cmd/kubeconfigwriter/main.go]: What happens if this is an empty string?  Does the field end up in the kubeconfig context with an empty string or is it omitted from the file completely?

Comment (cappyzawa) [reply cmd/kubeconfigwriter/main.go]: It's omitted from the file completely if this is empty.
I'll add a comment to code.

File: docs/resources.md (hunk 1 above)
Comment (cappyzawa) [reply docs/resources.md]: Thanks for your suggestion!

**Comment by cappyzawa** [comment]
@sbwsg I also added `namaspace` param to some test cases.
Could you review again if you have time, please:pray:

**Pull request merged**